lib/SPIRV/SPIRVLowerMemmove.cpp

```diff
@@ -63,34 +63,31 @@ class SPIRVLowerMemmoveBase {
     if (isa<PHINode>(Src))
       report_fatal_error("llvm.memmove of PHI instruction result not supported",
                          false);
-    auto *SrcTy = Src->getType();
-    auto *Length = cast<ConstantInt>(I.getLength());
-    auto *S = Src;
     // The source could be bit-cast or addrspacecast from another type,
     // need the original type for the allocation of the temporary variable
-    while (isa<BitCastInst>(S) || isa<AddrSpaceCastInst>(S))
-      S = cast<CastInst>(S)->getOperand(0);
-    SrcTy = S->getType();
+    auto *SrcTy = Src->stripPointerCasts()->getType();
+    auto *Length = cast<ConstantInt>(I.getLength());
     MaybeAlign Align = I.getSourceAlign();
     auto Volatile = I.isVolatile();
     Value *NumElements = nullptr;
     uint64_t ElementsCount = 1;
     if (SrcTy->isArrayTy()) {
-      NumElements = Builder.getInt32(SrcTy->getArrayNumElements());
       ElementsCount = SrcTy->getArrayNumElements();
+      NumElements = Builder.getInt32(ElementsCount);
     }
-    if (((ElementsCount > 1) && (Mod->getDataLayout().getTypeSizeInBits(
```

Comment:
The check was removed completely. Length parameter has no restrictions in LangRef.